wagtail/documents/views/bulk_actions/dispatcher.py

```diff
@@ -0,0 +1,14 @@
+from django.http.response import Http404
+
+from wagtail.core import hooks
+
+
+def index(request, action):
+    bulk_actions = hooks.get_hooks('register_document_bulk_action')
```

Comment:
suggestion: the body of this method is similar to that for pages, except for the hook. Maybe move it to base as 

```python
# admin/views/bulk_actions/dispatcher.py
def bulk_actions_index(request, action, hook):
    bulk_actions = hooks.get_hooks(hook)
    for action_func in bulk_actions:
        _action = action_func(request)
        if _action.action_type != action:
            continue
        return _action.dispatch(request)

    return Http404()
```

and then this could be something like

```python
from wagtail.admin.views.bulk_actions.dispatcher import bulk_actions_index

def index(request, action):
    return bulk_actions_index(request, action, "register_document_bulk_action")
```

ditto for the pages dispatcher

Reply:
I was thinking of maybe addressing this in a separate pr. I don't want this pr to contain refactoring commits which are not directly related to documents bulk actions functionality. The action-bar pr did become quite large because of that. Is that fine?

Reply:
sounds like a plan!